opm/autodiff/NonlinearSolver_impl.hpp

```diff
@@ -137,7 +137,11 @@ namespace Opm
             }
             converged = report.converged;
             linIters += report.linear_iterations;
-            wellIters += report.well_iterations;
+            if (report.well_iterations != std::numeric_limits<int>::min()) {
+                wellIters += report.well_iterations;
+            } else {
+                wellIters = report.well_iterations;
```

Comment:
> if `solveWellEq()` is not called, it will stay the same, I think.

Sure, but that's not the point I was trying to make.  A few [lines down](https://github.com/qilicun/opm-simulators/blob/bb0164f39f2979549dbac75b66622a24ed71260b/opm/autodiff/NonlinearSolver_impl.hpp#L157) from this loop we find the statement
```C++
wellIterations_ += wellIters; 
```
This is undefined behaviour (integer overflow) when both of these conditions are true

* `wellIters == std::numeric_limits<int>::min()`
*  `wellIterations_ < 0`

In other words, if we assign `wellIters = std::numeric_limits<int>::min()` as this patch does in the `else` branch then we need to take special care in the `wellIters_` update lest the program be undefined.